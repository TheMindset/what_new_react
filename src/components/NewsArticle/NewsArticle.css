article {
  background: brown;
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 10px;
}

a {
  border-bottom: 1px solid black;
  color: black;
  background-color: coral;
  border-radius: 0 0 5px 5px;
}

img {
  width: 100%;
}

.image {
  border-radius: 5px 5px 0 0;
  height: 45%;
  overflow: hidden;
  width: 100%;
}