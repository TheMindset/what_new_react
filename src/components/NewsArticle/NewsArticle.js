import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({story}) => {
  return [
    <article>
      Article
      <img src={story.img} alt=""/>
      <h2> {story.headline} </h2>
      <p> {story.description} </p>
      <a href={story.url} target="blank"> More informations here</a>
    </article>
  ]
}
export default NewsArticle;