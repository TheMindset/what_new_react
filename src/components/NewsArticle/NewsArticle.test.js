import React from 'react'
import { shallow } from 'enzyme'
import NewsArticle from './NewsArticle'

describe('NewsArticle', () => {

  test('should match snapshoot with all data passed correctly', () => {
    const article = { 
      img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
      headline: 'TEST TEST',
      description: 'Test test'
    }

    const wrapper = shallow(<NewsArticle 
      article={article}
    />)
    
    expect(wrapper).toMatchSnapshot()
  })
  
})
