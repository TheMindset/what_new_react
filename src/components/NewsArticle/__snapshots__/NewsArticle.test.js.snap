// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match snapshoot with all data passed correctly 1`] = `
<article>
  Article
  <div
    className="image"
  >
    <img
      alt=""
      src="https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp"
    />
  </div>
  <h3>
     
    TEST TEST
     
  </h3>
  <p>
     
    Test test
     
  </p>
  <a
    target="blank"
  >
     More informations here
  </a>
</article>
`;
